src/game/SpellEffects.cpp

```diff
@@ -1332,12 +1273,12 @@ void Spell::EffectTriggerSpell(SpellEffectIndex eff_idx)
         }
         // Terrordale Haunting Spirit #2, special case because chance is set to 101% in DBC while description is 55%
         case 23209:
-            if (urand(0, 100) < 55)
+            if (urand(0, 20) < 11)
                 m_caster->CastSpell(unitTarget, triggered_spell_id, true);
             return;
         // Terrordale Haunting Spirit #3, special case because chance is set to 101% in DBC while description is 35%
         case 23253:
-            if (urand(0, 100) < 35)
+            if (urand(0, 20) < 7)
```

Comment:
Actually urand is something defined in the cmangos sources (in src/shared/Util.{h,cpp}). It always returns an uint32, so urand(0,1) will either return 0 or 1, as an uint32 (you know that's not the same as a bool, but it's convertible to bool just like any integer type).

Internally, in the implementation, the numbers are generated using a [Mersenne Twister](https://en.wikipedia.org/wiki/Mersenne_Twister). The range doesn't really matter performance wise and you won't really notice a difference between urand(0,1) and urand(0,100000).